src/enclosure.c

```diff
@@ -171,3 +184,154 @@ int enclosure_open(const struct enclosure_device *enclosure)
 
 	return fd;
 }
+
+static struct enclosure_device *find_enclosure(char *enclosure_id)
+{
+	struct enclosure_device *enclosure_device;
+	list_for_each(sysfs_get_enclosure_devices(), enclosure_device) {
+		if (strncmp(enclosure_device->dev_path, enclosure_id, PATH_MAX) == 0) {
+			return enclosure_device;
+		}
+	}
+	return NULL;
+}
+
+status_t find_block_device_by_enclosure_slot(struct enclosure_device *encl, int index, enum ibpi_pattern *status, struct block_device **device)
```

Comment:
shouldn't it be static?

Reply:
this routine does much more than simple find/get, it also returns current pattern.
Why we cannot return ses_slot here?
``` C
struct ses_slot *find_enclosure_slot_by_index (struct enclosure_device *encl, int index)
for (int i = 0; i < encl->slots_count; i++) 
		if (encl->slots[i].index == index) 
			return &encl->slots[i];
return NULL;
}
```
Then in caller we could do:
```C
struct ses_slot s_slot = find_enclosure_slot_by_index (encl, index)
if (!s_slot)
    return error;
status = encl->slots[i].ibpi_status;
device = locate_block_by_sas_addr(encl->slots[i].sas_addr);
if (!device)
    return error;
```

Let me know if I missed something because I can see that `find_block_device_by_enclosure_slot` is used once

Reply:
Done as requested